faros_event.sh

```diff
@@ -0,0 +1,470 @@
+#
+# (Required build fields)
+# -bs / --build_status <status>  | BUILD_STATUS (e.g. Success, Failed, etc)
+# -r / --repo <repo>             | REPOSITORY
+# --vcs_org <vcs_org>            | VCS_ORG_UID
+# --vcs_source <vcs_source>      | VCS_SOURCE
+#
+# Optional fields:
+# Flag                  | Env Var                  | Default
+# -----------------------------------------------------------------------------
+# -g / --graph <graph>  | FAROS_GRAPH              | "default"
+# -url <url>            | FAROS_API_URL            | https://prod.api.faros.ai
+#                       | ORIGIN                   | "Faros_Script_Event"
+#                       | SOURCE                   | "Faros_Script"
+#                       | START_TIME               | Now (e.g. 1549932180000)
+#                       | END_TIME                 | Now (e.g. 1549932180000)
+#
+# (Optional deployment fields)
+#                       | DEPLOYMENT_UID           | Random UUID
+#                       | APPLICATION_PLATFORM     | "NA"
+#                       | DEPLOYMENT_ENV_DETAIL    | ""
+#                       | DEPLOYMENT_STATUS_DETAIL | ""
+#                       | DEPLOYMENT_START_TIME    | START_TIME
+#                       | DEPLOYMENT_END_TIME      | END_TIME
+#
+# (Optional build fields)
+#                       | BUILD_UID                | Commit Sha
+#                       | BUILD_STATUS_DETAIL      | ""
+#                       | BUILD_START_TIME         | START_TIME
+#                       | BUILD_END_TIME           | END_TIME
+#
+# Optional Script Flags:
+# Flag          | Description
+# -----------------------------------------------------------------------------
+# --dry_run     | If present, the event will be printed instead of emitted.
+# --print_event | If present, the event will be printed.
+# --debug       | If present, helpful information will be printed.
+main() {
+    EMIT_TYPE=$1
+    shift
+
+    parseArgs "$@"  
+    validateInput
+
+    if ((debug))
+    then
+        echo "Faros url: $faros_api_url"
+        echo "Faros graph: $faros_graph"
+        echo "Dry run: $dry_run"
+        echo "Print event: $print_event"
+        echo "Debug: $debug"
+    fi
+
+    if [ $EMIT_TYPE = "deployment" ]
+    then
+        validateDeploymentInput
+        makeDeploymentEvent
+    elif [ $EMIT_TYPE = "build" ]
+    then
+        validateBuildInput
+        makeBuildEvent
+    elif [ $EMIT_TYPE = "full" ]
+    then
+        validateBuildInput
+        validateDeploymentInput
+        makeFullEvent
+    else
+        echo "Unrecognized emit type: $EMIT_TYPE"
+        echo "Valid types: deployment, build, full"
+        exit 1
+    fi
+
+    if (($print_event)) || (($dry_run))
+    then
+        echo $request_body | jq .
+    fi
+
+    if !(($dry_run))
+    then
+        emitToFaros
+    else
+        echo "Dry run: Event NOT emitted to Faros"
+    fi
+    exit 0
+}
+
+function parseArgs() {
```

Comment:
isn't there a ready-to-use arg parsing function we could use?

Reply:
Here is a pretty good [page](https://stackoverflow.com/questions/192249/how-do-i-parse-command-line-arguments-in-bash) on the various ways to parse args in bash that I was referencing. You might be thinking of `getopts` which is great, other than its one major disadvantage which is it cannot parse flags other than a single character (i.e. `-f` can parse, `--flag` cannot parse). Let me know if you want to switch to another approach!

Reply:
Ah, bummer. Yeah I recall using getopts in the past. OK, let's keep it your way.